theano/sandbox/gpuarray/conv.cu

```diff
@@ -195,46 +188,47 @@ PyGpuArray_conv_valid(const PyGpuArrayObject *img,
         //we pass by ceil_intdiv in case the out_len is not a multiple of nb_split, we want nb_split the number of iteration.
         while (ceil_intdiv(out_len,nb_split)*out_wid>max_threads_dim0)
             nb_split++;
-        dim3 threads(out_wid, ceil_intdiv(out_len,nb_split));
-
-        dim3 grid(nbatch, nkern);
-        int shared_size=(img_size + kern_size)*sizeof(float);
-        void (*f)(const float*, const float*, float*,
-                  int, int, int, int,
-                  int, int);
-
-#define CONV_PATCH_SPECIAL(kern_wid) \
-            if(threads.y==out_len) f=conv_patch<true,kern_wid,false>;\
-            else f=conv_patch<true,kern_wid,true>;
-
-        CONV_PATCH_SPECIAL(THEANO_KERN_WID);
-
-         f<<< grid, threads, shared_size>>>
-             (cuda_get_ptr(img), cuda_get_ptr(kern), cuda_get_ptr(out),
-              img_len, img_wid, kern_len, kern_wid, nkern, nstack);
-
-        cudaError_t sts = cudaGetLastError();
-        if (cudaSuccess == sts)
+        size_t threads_per_block[3] = {(size_t)out_wid,
+                               ceil_intdiv((size_t)out_len,(size_t)nb_split),
+                               (size_t)1};
+        size_t n_blocks[3] = {(size_t)nbatch, (size_t)nkern, (size_t)1};
+
+        size_t shmem_sz = (img_size + kern_size)*sizeof(float);
+
+        GpuKernel *k = NULL;
+        if(threads_per_block[1]==out_len) k=&conv_patch_2_node_<<<<HASH_PLACEHOLDER>>>>_0;
+        else k=&conv_patch_3_node_<<<<HASH_PLACEHOLDER>>>>_0;
```

Comment:
I will say that I am not too keen on directly using `<<<<<HASH_PLACEHOLDER>>>>>` even though it looks like there is no other solution here.

Maybe you could use the facilities of COp to remove its use? If that looks too complex, then we can keep it as-is for now.